### PR TITLE
feat(azure): setup automatic deployment for dev

### PR DESCRIPTION
**Pull request by olros**
## Proposed changes
Removed Pipenv from Dockerfile and uses only pip with a requirements.txt file instead.
Setup a Github Action which builds and pushes the docker-image to Azure Conatiner Registry (ACI) on push to dev. The image is then deployed to Azure App Service and hosted. The docker-builds are cached which means that new deployments with no new dependencies takes about a minute. When there's new dependencies, deployment takes ca. 6 minutes.

*Will write docs for Azure on Slab*.

Issue number: closes #

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] CHANGELOG.md has been updated. [Guide](https://tihlde.slab.com/posts/changelog-z8hybjom)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Pull request title follows [conventional commits](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) (`type(scope): description`)
- [ ] Build (`make PR`) was run locally without failures

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,53 @@
+psycopg2
```
Comment (Eirsteir) [inline requirements.txt:1]: Perhaps we should pin these?

**Review by Eirsteir — COMMENTED** [review]
Looks good! Would like to have someone else take a look at this as well :)

File: .github/workflows/deploy_dev.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        push: true
+        tags: leptondevregistry.azurecr.io/lepton:${{ github.sha }}
+        cache-from: type=local,src=/tmp/.buildx-cache
+        cache-to: type=local,dest=/tmp/.buildx-cache
+
+    - name: Deploy to Azure Web App
+      uses: azure/webapps-deploy@v2
+      with:
+        app-name: 'lepton-dev-api'
+        slot-name: 'production'
```
[the remark this answers is not in the document]
Comment (olros) [reply .github/workflows/deploy_dev.yml:50]: Production betyr bare at det er denne som kjøres i app servicen

File: requirements.txt (hunk 1 above)
Comment (olros) [reply requirements.txt:1]: Opp til deg 🤷‍♂️

File: Dockerfile [hunk 3]
```diff
@@ -4,22 +4,23 @@ ENV PYTHONUNBUFFERED 1
...
-    # Install required linux packages (remove postgresql-dev when Heroku isn't needed)
-    apk add --no-cache mariadb-dev postgresql-dev
+    apk add --no-cache mariadb-dev
 
-COPY Pipfile Pipfile.lock ./
+COPY requirements.txt ./
 
 # Install required python packages
 RUN apk --update add --no-cache --virtual .build-deps gcc musl-dev python3-dev libffi-dev openssl-dev cargo py-pip linux-headers libc-dev build-base && \
       pip install --upgrade pip && \
```
Comment (Eirsteir) [inline Dockerfile:13]: Bare av nysgjerrighet, oppdaterer vi pip både her og på linje 7? `pip install --upgrade pip`

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+sentry-sdk
+celery == 5.0.5
+azure-storage-blob == 12.8.0
+
+# Django
+# ------------------------------------------------------------------------------
+django-enumchoicefield
+django-dotenv
+django-filter
+django-heroku
```
Comment (Eirsteir) [inline requirements.txt:15]: Tror vi kan fjerne denne siden vi går vekk fra heroku eller er det for tidlig? https://pypi.org/project/django-heroku/

File: requirements.txt [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+django-polymorphic ~= 3.0
+django-rest-polymorphic
+
+django-mptt
+
+# Documentation
+# ------------------------------------------------------------------------------
+drf-yasg
+
+# Code quality
```
Comment (Eirsteir) [inline requirements.txt:31]: Strengt tatt burde alle dependencies under her bli lagt i en separat requirements.txt som kun blir brukt lokalt. Mulig å hente inspirasjon [herfa](https://github.com/pydanny/cookiecutter-django/tree/master/%7B%7Bcookiecutter.project_slug%7D%7D/compose). 

Kan alltids være en fremtidig forbedring 😉

File: docker-compose.dev.yml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+version: '3'
+
+# Config file for registry settings in Azure App Service dev
```
Comment (Eirsteir) [inline docker-compose.dev.yml:3]: Hvordan blir denne brukt av Azure egentlig?

**Review by Eirsteir — APPROVED** [review]
Ser veldig bra ut! Har bare et par generelle spørsmål og bemerkninger 😄

File: Dockerfile (hunk 3 above)
Comment (olros) [reply Dockerfile:13]: Ser sånn ut ja 😅 Vet helt ærlig ikke om det er nødvendig, men tar så lang tid med make fresh når man endrer så har ikke orket å prøve å endre.

File: requirements.txt (hunk 4 above)
Comment (olros) [reply requirements.txt:15]: Prøvde det, men den inneholder noen subdependencies som gir bedre hosting av static sites og tilkoblig til database og litt sånt utils som er kjekt å ha

File: docker-compose.dev.yml (hunk 6 above)
Comment (olros) [reply docker-compose.dev.yml:3]: La fila her for at man skal ha den et sted. Det er den som legges inn i App Service config for å spinne opp både Django og Celery.

File: requirements.txt (hunk 5 above)
Comment (olros) [reply requirements.txt:31]: Lager et issue 😁

**Pull request merged**